### PR TITLE
[INT] Absences API: filtering absences by updated_at

### PR DESCRIPTION
**Pull request by quangnguyen30192**

### CODE REVIEW

File: personio-personnel-data-api-oa3.yaml [hunk 1]
```diff
@@ -1095,6 +1113,9 @@ paths:
                           certificate:
                             status: not-required
                           created_at: '2017-01-17T10:32:18+0100'
+                          updated_at: '2017-01-17T10:32:18+0100'
+                    offset: 0,
+                    limit: 200,
```
Comment (amureki) [inline personio-personnel-data-api-oa3.yaml:1118]: Greetings fellows!

As I understood, this change just was implemented on your server side just this morning?
Can we have an updated [CHANGELOG](https://developer.personio.de/changelog)? Or how can we get such breaking changes being announced before?

We at [@thermondo](https://github.com/Thermondo/) have a critical production application broken due to the new limitation, which we were not aware of.
We are working on adding pagination on our side right now, but all our trials are also showing that there are some things breaking on your side:


```
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://api.personio.de/v1/company/time-offs?limit=200&offset=38
```

There was an 500 from `auth` endpoint, if we are calling it fast, but I lost logs for this, but I can append it later, if needed.


Best,
Rust

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: Hi Rust, 

Thank you for informing us about your situation. We're updating the changelog today.

> There was an 500 from auth endpoint, if we are calling it fast and often

This should be `Too many attemps` error as we have a limit 50 hits for every 2 minutes. You may have a mechanism to wait and retry when running into this error.

Comment (amureki) [reply personio-personnel-data-api-oa3.yaml:1118]: Hey Quang Nguyen,

thanks for the response!

>If you need to get the full list of items at once, then the API should only return the results with pagination when limit or offset is specified on the query, otherwise, the full list of items should be returned.

I think, there is a something wrong, as what is written here is different for me with real API I am working on right now.

https://developer.personio.de/reference#get_company-time-offs endpoint returns me only 200 records when I am not adding any query params. So, I can clearly say, there is a default value set on your side, which was not there two days ago.
If this breaking behaviour would not change, nothing would even break on our side, because it'd work the say way as before. But right now, you are setting a default and there is no way to query more than 200 records, so it is clearly a breaking change in API, which should not happen in the same `v1` version.

So, to recap:

1. 
```
curl --request GET \
  --url 'https://api.personio.de/v1/company/time-offs' \
  --header 'Accept: application/json'
```
returns 200 items, even if I have 30000 records.


2.
```
curl --request GET \
  --url 'https://api.personio.de/v1/company/time-offs?limit=201&offset=0' \
  --header 'Accept: application/json'
```

returns `HTTP 422: Unprocessable Entity for url: https://api.personio.de/v1/company/time-offs?limit=201&offset=0`


I am happy to provide more examples, if something is still not clear.


Cheers,
Rust

Comment (amureki) [reply personio-personnel-data-api-oa3.yaml:1118]: Also, changelog entity does not reflect this breaking limitation at all: https://developer.personio.de/changelog/absence-api-filter-by-updated_at

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: Hi Rust,

We're deploying a fix for it. the API would return all of the items at once if you don't include `limit` and `offset` in the query. 

https://developer.personio.de/changelog/breaking-change-absence-pagination-fix-is-underway-1

I will let you know when it's done.

Sorry for the inconvenience.

Thanks,
Regards

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: Hi Rust,

The issue has been fixed. Can you please try again?

Thanks and best regards,

Comment (amureki) [reply personio-personnel-data-api-oa3.yaml:1118]: Hello,

I just tried the "original" request, it took some time, but returned me all the results. I hope, this will remain. 👍

On a sidenote: now "paginated" endpoints seem to be not returning "metadata" any more, but only "data":
image

So there is no easy way to determine total pages and total elements if we are using pagination.

Could you, please, confirm, that you are still doing some changes on production API, and it is not stable yet?

Best,
Rust

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: Hi Rust,

Currently, with offset = 0 then the pagination metadata does not work, e.g `https://api.personio.de/v1/company/time-offs?limit=200&offset=0`.

We're deploying a fix for it.

Regards,
Quang

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: It should work for now. 

Thanks and best regards,

Comment (amureki) [reply personio-personnel-data-api-oa3.yaml:1118]: @quangnguyen30192 lovely, thank you very much for support and fast pace 🚀 

Best,
Rust

Comment (quangnguyen30192) [reply personio-personnel-data-api-oa3.yaml:1118]: You're very welcome 🎉

Regards,
Quang
